config/initializers/hypershield.rb

```diff
@@ -0,0 +1,41 @@
+# Specify which environments to use Hypershield
+# Optional to setup, can help shield sensitive data for
+# tools like Blazer.
+# further setup instructions: https://github.com/ankane/hypershield#database-setup
+
+# Validate that hypershield schema exists before trying to use it
+sql = <<-SQL
+  SELECT schema_name FROM information_schema.schemata WHERE schema_name = 'hypershield';
+SQL
+schema_exists = ActiveRecord::Base.connection.execute(sql).map(&:itself).count == 1
+
+if Rails.env.production? && schema_exists
```

Comment:
A couple of ideas here.

As this is only available in production, we don't need to check the schema everytime we boot the app in development, so we should do something like:

```ruby
if production
  if schema exists
  end
end
```

Also you could simplify the schema check with: 

```ruby
ActiveRecord::Base.connection.schema_exists?("hypershield")
```

Reply:
Ah thank you for the shortcut and good call not always checking it!